data-prepper-plugins/elasticsearch/src/main/java/com/amazon/dataprepper/plugins/sink/elasticsearch/BulkRetryStrategy.java

```diff
@@ -27,14 +32,23 @@
 
     private final RequestFunction<BulkRequest, BulkResponse> requestFunction;
     private final BiConsumer<DocWriteRequest<?>, Throwable> logFailure;
+    private final PluginMetrics pluginMetrics;
     private final Supplier<BulkRequest> bulkRequestSupplier;
 
+    private final Counter sentDocumentsCounter;
+    private final Counter documentErrorsCounter;
+
     public BulkRetryStrategy(final RequestFunction<BulkRequest, BulkResponse> requestFunction,
                              final BiConsumer<DocWriteRequest<?>, Throwable> logFailure,
+                             final PluginMetrics pluginMetrics,
                              final Supplier<BulkRequest> bulkRequestSupplier) {
         this.requestFunction = requestFunction;
         this.logFailure = logFailure;
+        this.pluginMetrics = pluginMetrics;
         this.bulkRequestSupplier = bulkRequestSupplier;
+
+        sentDocumentsCounter = pluginMetrics.counter(DOCUMENTS_SUCCESS);
```

Comment:
So we have 2 metrics here: 1 for errors and one for successes. Do we have (or want) any notion of keeping track of how many requests failed and then succeeded on retry?

Reply:
Fair point. This should be a good metric to tell whether there could be too many requests for ES to handle. I think we can introduce a success-on-first-attempt counter to distinguish.

Reply:
Sure, maybe just a todo

Reply:
Done in latest commits